src/main/webapp/app/quiz/re-evaluate/quiz-re-evaluate-warning.controller.js

```diff
@@ -61,41 +61,96 @@
+                                        vm.questionElementInvalid = true;
+                                    }
+                                    //answer correctness changed?
+                                    if (answer.isCorrect !== backUpAnswer.isCorrect) {
+                                        vm.questionCorrectness = true;
+                                    }
+                                }
+                            }
+                        });
+                    }
+                    //check DragAndDropQuestions
+                    if (question.type === "drag-and-drop") {
+                        if (question.dragItems.length !== backUpQuestion.dragItems.length
+                            || question.dropLocations.length !== backUpQuestion.dropLocations.length) {
+                            vm.questionElementDeleted = true;
+                        }
+                        if (!angular.equals(question.correctMappings, backUpQuestion.correctMappings)) {
+                            vm.questionCorrectess = true;
+                        }
+
+                        //check each dragItem
+                        question.dragItems.forEach(function (dragItem) {
+                        // only check if there are no changes on the question-elements yet
+                            if (!vm.questionElementInvalid) {
+                                var backUpDragItem = null;
+                                for (var j = 0; backUpDragItem === null && j < backUpQuestion.dragItems.length; j++) {
+                                    if (backUpQuestion.dragItems[j].id === dragItem.id) {
+                                        backUpDragItem = backUpQuestion.dragItems[j];
+                                    }
+                                }
+                                if (backUpDragItem !== null) {
+                                    //dragItem set invalid?
+                                    if (dragItem.invalid !== backUpDragItem.invalid) {
+                                        vm.questionElementInvalid = true;
+                                    }
                                 }
                             }
-                            if (backUpAnswer != null) {
-                                //answer set invalid?
-                                if (answer.invalid !== backUpAnswer.invalid) {
-                                    vm.answerInvalid = true;
+                        });
+
+                        //check each dropLocation
+                        question.dropLocations.forEach(function (dropLocation) {
+                            // only check if there are no changes on the question-elements yet
+                            if (!vm.questionElementInvalid) {
+                                var backUpDropLocation = null;
+                                for (var j = 0; backUpDropLocation === null && j < backUpQuestion.dropLocations.length; j++) {
```

Comment:
can you please use a for each loop here?

Reply:
@MoritzIssig Again, take a look at the find() function in javascript arrays: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find